docs/messaging/kafka-integration.md

```diff
@@ -2,94 +2,204 @@
+
+## Client integration
+
+To get started with the .NET Aspire Apache Kafka integration, install the [📦 Aspire.Confluent.Kafka](https://www.nuget.org/packages/Aspire.Confluent.Kafka) NuGet package in the client-consuming project, that is, the project for the application that uses the Apache Kafka client.
 
 ### [.NET CLI](#tab/dotnet-cli)
 
 ```dotnetcli
-dotnet add package Aspire.Hosting.Kafka
+dotnet add package Aspire.Confluent.Kafka
 ```
 
 ### [PackageReference](#tab/package-reference)
 
 ```xml
-<PackageReference Include="Aspire.Hosting.Kafka"
+<PackageReference Include="Aspire.Confluent.Kafka"
                   Version="[SelectVersion]" />
 ```
 
 ---
 
-In your app host project, register a Kafka container and consume the connection using the following methods:
+### Add Kafka producer
+
+In the _:::no-loc text="Program.cs":::_ file of your client-consuming project, call the <xref:Microsoft.Extensions.Hosting.AspireKafkaProducerExtensions.AddKafkaProducer%2A> extension method to register an `IProducer<TKey, TValue>` for use via the dependency injection container. The method takes two generic parameters corresponding to the type of the key and the type of the message to send to the broker. These generic parameters are used to new an instance of `ProducerBuilder<TKey, TValue>`. This method also takes connection name parameter.
 
 ```csharp
-var builder = DistributedApplication.CreateBuilder(args);
+builder.AddKafkaProducer<string, string>("messaging");
+```
 
-var messaging = builder.AddKafka("messaging")
-                       .WithKafkaUI();
+You can then retrieve the `IProducer<TKey, TValue>` instance using dependency injection. For example, to retrieve the producer from an `IHostedService`:
 
-var myService = builder.AddProject<Projects.MyService>()
-                       .WithReference(messaging);
+```csharp
+internal sealed class Worker(IProducer<string, string> producer) : BackgroundService
+{
+    // Use producer...
+}
 ```
 
-The `WithKafkaUI()` extension method which provides a web-based interface to view the state of the Kafka container instance. The `WithReference` method configures a connection in the `MyService` project named `messaging`. In the _:::no-loc text="Program.cs":::_ file of `MyService`, the Apache Kafka broker connection can be consumed using:
+For more information on workers, see [Worker services in .NET](/dotnet/core/extensions/workers).
+
+### Add Kafka consumer
+
+To register an `IConsumer<TKey, TValue>` for use via the dependency injection container, call the <xref:Microsoft.Extensions.Hosting.AspireKafkaConsumerExtensions.AddKafkaConsumer%2A> extension method in the _:::no-loc text="Program.cs":::_ file of your client-consuming project. The method takes two generic parameters corresponding to the type of the key and the type of the message to receive from the broker. These generic parameters are used to new an instance of `ConsumerBuilder<TKey, TValue>`. This method also takes connection name parameter.
```

Comment:
Same thing as I mentioned in AddKafkaProducer about the clause "to new an instance"